spec/features/users/user_signs_in_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "rails_helper"
+
+feature 'user signs in', %{
+  As a user
+  I want to sign in
+  So that I can post new bourbons or review existing bourbons
+
+  Acceptance Criteria
+  [x] I am authenticated and gain access to the system if I specify valid,
+      previously registered username and password.
+  [x] I remain unathenticated if I specify invalid username/password
+  [x] I cannot sign in again if I am already signed in
+} do
+
+  scenario 'existing user specifies a valid username and password' do
+    user = FactoryGirl.create(:user)
+    visit root_path
+    click_link 'Sign In'
+    fill_in 'Username', with: user.username
+    fill_in 'Password', with: user.password
+    click_button 'Sign In'
+    expect(page).to have_content('Welcome back!')
+    expect(page).to have_content('Sign Out')
+  end
+
+  scenario 'a nonexistent username and password is supplied' do
+    visit root_path
+    click_link 'Sign In'
+    fill_in 'Username', with: 'anonexistentuser'
+    fill_in 'Password', with: 'whawhawha'
+    click_button 'Sign In'
+    expect(page).to_not have_content('Welcome back!')
+    expect(page).to_not have_content('Sign Out')
+    expect(page).to have_content('Invalid username or password')
+  end
+
+  scenario 'an existing username with wrong password is denied access' do
+    user = FactoryGirl.create(:user)
```

Comment:
Consider how these feature tests could be more DRY (using a `before` block, perhaps!)